test/mock/mock-management-plane-api.yaml

```diff
@@ -0,0 +1,170 @@
+openapi: 3.0.2
+info:
+  title: Management Plane API
+  version: 0.0.1
+security:
+  - Bearer: []
+paths:
+  /instances/{instanceId}/files/:
+    get:
+      summary: Return all files metadata for a data plane instance.
+      operationId: GetInstanceFilesMetadata
+      parameters:
+        - in: header
+          name: tenantId
+          schema:
+            type: string
+            format: uuid
+          required: false
+        - in: path
+          name: instanceId
+          schema:
+            type: string
+          required: true
+          description: Instance ID
+          example: "aecea348-62c1-4e3d-b848-6d6cdeb1cb9c"
+      responses:
+        '200':
+          description: Files metadata for a data plane instance.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FilesResponse'
+  /instances/{instanceId}/files/{filePath}:
+    get:
+      summary: Download a specific file for a data plane instance.
+      operationId: GetInstanceFile
+      parameters:
+        - in: header
+          name: tenantId
+          schema:
+            type: string
+            format: uuid
+          required: false
+        - in: path
+          name: instanceId
+          schema:
+            type: string
+          required: true
+          description: Instance ID
+          example: "aecea348-62c1-4e3d-b848-6d6cdeb1cb9c"
+        - in: path
+          name: filePath
+          schema:
+            type: string
+          required: true
+          description: file location
+          example: "/etc/nginx/nginx.conf"
+        - in: query
+          name: version
+          schema:
+            type: string
+          required: false
+          description: version ID
+          example: ".WC2wyv_tB8kZpc87emyUPWi7BedaRSs"
+        - in: query
+          name: encoded
+          schema:
+            type: boolean
+          required: true
+          description: file content encoded
+      responses:
+        '200':
+          description: File downloaded
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/DownloadResponse'
+                  - $ref: '#/components/schemas/DownloadResponseEncoded'
+                discriminator:
+                  propertyName: type
+
+components:
```

Comment:
Just pointing out that we haven't marked many of our components' fields as `required`, meaning that the generated structs' fields will be pointers. I think it's good practice to mark fields that we expect to be populated as `required` to reduce the need for nil checks in our code.